src/test/java/elements/SportCategoryElements.java

```diff
@@ -0,0 +1,118 @@
+    private static final String DESCRIBE_COLOR_FIELD = "h4.text-sm";
+    private static final String SIZE_BTN = "//span[text()='Size']";
+    private static final String COLOR_BTN = "//button[@data-test-id='undefined-btn']/span[text()='Color']";
+
+    public static String priceXpath(int first, int second) {
+        final String xpathPrice = String.format("//input[@id='(%d..%d)']/../div",first,second);
+        return xpathPrice;
+    }
+
+    protected WebElement getPriceFromToElement(int first, int second) {
+        return waitUntilElementToBeClickableByXpath(priceXpath(first,second));
+    }
+
+    public WebElement getTeamSelectBtn() {
+        return waitUntilElementToBeVisibleByXpath(TEAM_SELECT_BTN);
+    }
+
+    public WebElement getBMWMotorsportCheckBox() {
+        return waitUntilElementToBeVisibleByXpath(BMW_MOTORSPORT_CHECKBOX);
+    }
+
+    protected WebElement getFilterBtnElement() {
+        return waitUntilElementToBeVisibleByXpath(FILTER_BTN);
+    }
+
+    protected WebElement getListItemsName() {
+        return waitUntilElementToBeVisibleByXpath(LIST_ITEMS_NAME);
+    }
+
+    protected WebElement get20_OffCloseBtn() {
+        return waitUntilElementToBeVisibleByXpath(GET_20_OFF);
+    }
+
+    protected WebElement getCloseFilterBtn() {
+        return waitUntilElementToBeVisibleByCss(CLOSE_FILTER_BTN);
+    }
+
+    protected WebElement getGenderBtn() {
+        return waitUntilElementToBeVisibleByXpath(GENDER_BTN);
+    }
+
+    protected WebElement getBlackColorBtn() {
+        return waitUntilElementToBeVisibleByCss(BLACK_COLOR_BTN);
+    }
+
+    protected WebElement getMenCheckBox() {
+        return waitUntilElementToBeVisibleByXpath(MEN_CHECKBOX);
+    }
+
+    protected WebElement getCategoryBtn() {
+        return waitUntilElementToBeVisibleByXpath(CATEGORY_BTN);
+    }
+
+    public WebElement colorBtnElement() {
+        return waitUntilElementToBeVisibleByXpath(COLOR_BTN);
+    }
+
+    public WebElement getClothingElement() {
+        return waitUntilElementToBeClickableByXpath(CLOTHING_CHECKBOX);
+    }
+
+    public WebElement getPriceElement() {
+        return waitUntilElementToBeClickableByXpath(PRICE_BTN);
+    }
+
+    public WebElement sizeElement() {
+        return waitUntilElementToBeClickableByXpath(SIZE_BTN);
+    }
+
+    public WebElement describeColorElement() {
+        return waitUntilElementToBeClickableByCss(DESCRIBE_COLOR_FIELD);
+    }
+
+    public List<WebElement> listSalePriceElements() {
+        return waitUntilPresenceOfAllElementsByCss(SALE_PRICE);
+    }
+
+    public List<WebElement> listFullPriceElements() {
+        return waitUntilPresenceOfAllElementsByXpath(FULL_PRICE);
+    }
+
+    public WebElement pumaLogo() {
+        return waitUntilElementToBeVisibleByXpath(PUMA_LOGO);
+    }
+
+    public List<WebElement> listProductNames() {
+        return listWaitUntilElementsToBeVisibleByXpath(LIST_ITEMS_NAME);
+    }
+}
```

Comment:
83-118 lines too..please use incapsulation OOP principle)

Reply:
it's not a bug it's a feature :))
Because these "public WebElements" must be used out of this Class and it's inheritors.
I use them just as a WebElement particular in the Test without any 
actions (for example:  click, get text, send keys, etc)